packages/multi-test/src/custom_handler.rs

```diff
@@ -0,0 +1,60 @@
+use cosmwasm_std::{Addr, Api, Binary, BlockInfo, Empty, Storage};
+use mockall::automock;
+
+use anyhow::Result as AnyResult;
+
+use crate::AppResponse;
+
+/// Custom message handler trait. Implementator of this trait is mocking environment behaviour on
+/// given custom message.
+pub trait CustomHandler<ExecC = Empty, QueryC = Empty> {
+    fn execute(
+        &self,
+        api: &dyn Api,
+        storage: &mut dyn Storage,
+        block: &BlockInfo,
+        sender: Addr,
+        msg: ExecC,
+    ) -> AnyResult<AppResponse>;
+
+    fn query(
+        &self,
+        api: &dyn Api,
+        storage: &dyn Storage,
+        block: &BlockInfo,
+        msg: QueryC,
+    ) -> AnyResult<Binary>;
+}
+
+/// Simplified version of `CustomHandler` having only arguments which are not app internals - they
+/// are just discarded. Usefull for simpler mocking.
+#[automock(type QueryResult = Binary;)]
+pub trait SimpleCustomHandler<ExecC = Empty, QueryC = Empty> {
+    type QueryResult;
+
+    fn execute(&self, block: &BlockInfo, sender: Addr, msg: ExecC) -> AnyResult<AppResponse>;
+    fn query(&self, block: &BlockInfo, msg: QueryC) -> AnyResult<Binary>;
+}
+
+impl<ExecC, QueryC, T: SimpleCustomHandler<ExecC, QueryC>> CustomHandler<ExecC, QueryC> for T {
```

Comment:
I would just have this implementation return error (so you can test it works). For any real usage, we will have time to build a module to define the semantics of this. We will want to mock out the real blockchain behaviour, so I would tend against one-off mocks.

Reply:
Or maybe a FailingCustomHandler, which returns a pre-defined error (set in new?)
And SuccessCustomHandler, which returns a fixed response (set in new?)

One or both just to help in your testing in other packages, but those don't even have to be public outside of this crate.